tests/gis_tests/geoapp/tests.py

```diff
@@ -403,6 +403,21 @@ def test_null_geometries(self):
         State.objects.filter(name='Northern Mariana Islands').update(poly=None)
         self.assertIsNone(State.objects.get(name='Northern Mariana Islands').poly)
 
+    @skipUnlessDBFeature("supports_null_geometries", "supports_crosses_lookup", "supports_relate_lookup")
+    def test_null_geometries_lookups(self):
+        " Testing that NULL features are not hit with spatial lookup functions "
+        null = State.objects.create(name='NULL', poly=None)
+        qs = State.objects.all()
+        self.assertNotIn(null, list(qs.filter(poly__intersects=LineString((0, 0), (1, 1), (5, 5)))))
```

Comment:
You could use a loop and subTest to make the test less repetitive. e.g. have a list of dicts like `{'poly__equals': Point(1, 1)}` and then use `qs.filter(**filter_kwarg)` within the loop.